### PR TITLE
#169817539 User account verify

### PR DESCRIPTION
**Pull request by higustave-ops**
#### What does this PR do?
`Create user account verify feature`
#### Description of Task to be completed?
`To create a user account verification feature by allowing a user to be able to verify his/her account after a successful signup. A signing up user should copy the token sent to his email and then paste it into Headers Authorization of POSTMAN in order to be able to verify his account`
#### How should this be manually tested?
* Clone this repo
* Navigate in the repo root using `CMD` or any `TERMINAL`
* Install all packages using `npm install`
* Type `npm run dev` for Postman testing And `npm run test` for mocha testing
* Open postman choose POST and type `localhost:3000/api/auth/signup`
* Click `body` then `x-www-form-urlencoded`
* Fill in `firstName`, `lastName`, `email`, `password` and their respective values (`Key: Value`)pair
* Click `SEND` on the top right corner of Postman
* Go to your email (The one you signed up with)
* Copy the token provided in it
* In postman choose PATCH and type `localhost:3000/api/auth/account_verify`
* Paste it (token) into POSTMAN's Headers-->Authorization field
* Click send button, and once the token is not expired, your account gets verified.

#### What are the relevant pivotal tracker stories?
- [#169817539](https://www.pivotaltracker.com/story/show/169817539)

#### Screenshots (if appropriate)
`Postman screenshot Upon successful user account verification`
image

`Postman screenshot once provided token has expired`
image

`Postman screenshot once trying to verify an already verified account`
image

`Test screenshot`
image

#### Questions:

### CODE REVIEW

File: src/controllers/auth.controller.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
+import { config } from 'dotenv';
 import ResponseService from '../services/response.service';
 import UserService from '../services/user.service';
 import BcryptService from '../services/bcrypt.service';
 import JwtService from '../services/jwt.service';
 
+config();
```
Comment (victor-abz) [inline src/controllers/auth.controller.js]: why this config here?

File: src/routes/auth.route.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import express from 'express';
 import authController from '../controllers/auth.controller';
```
Comment (erickyvand) [inline src/routes/auth.route.js]: the import name should match the export name

File: src/controllers/auth.controller.js [hunk 3]
```diff
@@ -15,8 +17,8 @@ class AuthController {
  * @static
  * @description POST /api/v1/signUp
  * @param {object} req request object
- * @param {object} res ResponseService object
- * @memberof AuthController
+ * @param {object} res response object
+ * @memberof UsersController
```
Comment (victor-abz) [inline src/controllers/auth.controller.js]: Should be AuthController

File: src/controllers/auth.controller.js [hunk 4]
```diff
@@ -15,8 +17,8 @@ class AuthController {
  * @static
  * @description POST /api/v1/signUp
  * @param {object} req request object
- * @param {object} res ResponseService object
- * @memberof AuthController
+ * @param {object} res response object
```
Comment (victor-abz) [inline src/controllers/auth.controller.js:18]: Should be ResponseService

File: src/controllers/auth.controller.js [hunk 5]
```diff
@@ -35,13 +37,70 @@ class AuthController {
...
     return ResponseService.send(res);
   }
 
+  /**
+* users can verify his/her account
+* @static
+* @description PATCH /api/auth/account_verify
+* @param {object} req request object
+* @param {object} res response object
+* @memberof UsersController
```
Comment (victor-abz) [inline src/controllers/auth.controller.js]: AuthCOntroller instead of UsersController

**Review by gadishimwe — COMMENTED** [review]
LGTM

File: src/services/user.service.js [hunk 6]
```diff
@@ -31,6 +33,23 @@ class UserService {
...
 
+  /**
+   *
+   *
+   * @static
+   * @param {item} email
+   * @param {item} isVerified
+   * @returns {updated} @memberof UserService
+   */
+  static updateAccountStatus(email, isVerified) {
```
Comment (victor-abz) [inline src/services/user.service.js]: Use the same function update user below(the functionalities are the same) you don't need to create a new one.

File: src/services/user.service.js [hunk 7]
```diff
@@ -44,6 +63,24 @@ class UserService {
...
+   *
+   *
+   * @static
+   * @param {item} email
+   * @param {item} emailSubject
+   * @param {item} emailBody
+   * @param {item} token
+   * @returns {SendEmailService} @memberof UserService
+   */
+  static sendAccVerificationLink(email) {
```
Comment (victor-abz) [inline src/services/user.service.js]: Why is this function in services? does it deal with database? I thought services deals with db only.

Comment (victor-abz) [reply src/services/user.service.js]: you can put it may be in controller or...

Comment (victor-abz) [reply src/services/user.service.js]: check from eric

Comment (victor-abz) [reply src/services/user.service.js]: he was supposed to create a function like this. which you can call whenever you need to send email

File: src/services/user.service.js [hunk 8]
```diff
@@ -31,6 +33,23 @@ class UserService {
...
+   * @param {item} email
+   * @param {item} isVerified
+   * @returns {updated} @memberof UserService
+   */
+  static updateAccountStatus(email, isVerified) {
+    return Users.update({ isVerified }, {
+      where: {
+        email
+      },
+      logging: false
```
Comment (victor-abz) [inline src/services/user.service.js]: remove this line as it have been disabled in config

**Review by victor-abz — CHANGES_REQUESTED** [review]

File: src/controllers/auth.controller.js (hunk 3 above)
Comment (higustave-ops) [reply src/controllers/auth.controller.js]: @victor-abz this has been already resolved. I have got it as feedback from @erickyvand .

File: src/controllers/auth.controller.js (hunk 4 above)
Comment (higustave-ops) [reply src/controllers/auth.controller.js:18]: At which line exactly?

Comment (victor-abz) [reply src/controllers/auth.controller.js:18]: 20

File: src/controllers/auth.controller.js (hunk 3 above)
Comment (victor-abz) [reply src/controllers/auth.controller.js]: Line 21?

File: src/services/user.service.js (hunk 7 above)
Comment (higustave-ops) [reply src/services/user.service.js]: For this, i think the business logic must not reside in the controller

Comment (victor-abz) [reply src/services/user.service.js]: why?

File: src/controllers/auth.controller.js (hunk 3 above)
Comment (higustave-ops) [reply src/controllers/auth.controller.js]: Ok. worked on it. You can re review. I have finished your feedback.

File: src/services/user.service.js (hunk 7 above)
Comment (higustave-ops) [reply src/services/user.service.js]: Thats how it should be. I have researched on internet.

**Review by victor-abz — APPROVED** [review]

**Review by erickyvand — APPROVED** [review]

**Review by gadishimwe — APPROVED** [review]
LGTM

**Pull request merged**